.github/workflows/imagepuller-eval.yml

```
@@ -0,0 +1,48 @@
+name: imagepuller evaluate resource usage
+
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - imagepuller/**
+      - tools/imagepuller/**
+      - packages/by-name/imagepuller*/**
+  push:
+    branches:
+      - main
+
+jobs:
+  run-eval:
+    runs-on: ubuntu-24.04
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          persist-credentials: false
+      - uses: ./.github/actions/setup_nix
+        with:
+          githubToken: ${{ secrets.GITHUB_TOKEN }}
+          cachixToken: ${{ secrets.CACHIX_AUTH_TOKEN }}
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@b47578312673ae6fa5b5096b330d9fbac3d116df # v4.2.1
+        with:
+          role-to-assume: arn:aws:iam::795746500882:role/ContrastPublicBucketRW
+          aws-region: eu-central-1
+      - name: Download baseline JSON from S3
+        env:
+          S3_BUCKET_PATH: "imagepuller-eval"
+        run: |
+          aws s3 cp "s3://contrast-public/${S3_BUCKET_PATH}/baseline.json" "baseline.json"
+      - name: Run imagepuller-eval script
+        run: |
+          imagepullereval="$(nix build .#imagepuller-eval --print-out-paths)/bin/imagepuller-eval"
+          imagepuller="$(nix build .#imagepuller --print-out-paths)/bin/imagepuller"
+          sudo "$imagepullereval" "$imagepuller" -c baseline.json -o baseline-new.json
+      - name: Upload new baseline.json to S3
+        if: github.event_name != 'pull_request'
```

```suggestion
        if: github.event_name == 'push' && github.ref_name == 'main'
```
better to be specific here. Current condition won't exclude workflow_dispatch.